### PR TITLE
Implement `Domain` update code path

### PR DESCRIPTION
**Pull request by razor-1**
Issue #, if available:
https://github.com/aws-controllers-k8s/community/issues/1181

Description of changes:
This implements `Domain` update support for the opensearchservice controller.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: test/e2e/tests/test_domain.py
Comment (a-hilaly) [inline test/e2e/tests/test_domain.py:1]: Can we add a simple test case for updating a domain CR? either a new test or updating an existing one should be good.

**Review by a-hilaly — COMMENTED** [review]
This is great! Thank you so much @razor-1! I left one comment in-line

File: test/e2e/tests/test_domain.py
Comment (razor-1) [reply test/e2e/tests/test_domain.py:1]: yes, good call. Added.

**Review by a-hilaly — APPROVED** [review]
Thank you so much @razor-1 !

Can you please send a release PR? Just running this command will generate the artifacts :)

```
RELEASE_VERSION=v0.0.26 ./scripts/build-controller-release.sh opensearchservice
```


/lgtm

**Pull request merged**